backend/infrahub/core/migrations/graph/m027_delete_isolated_nodes.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Sequence
+
+from infrahub.core.migrations.shared import GraphMigration, MigrationResult
+from infrahub.log import get_logger
+
+from ...query import Query, QueryType
+
+if TYPE_CHECKING:
+    from infrahub.database import InfrahubDatabase
+
+log = get_logger()
+
+
+class DeleteIsolatedNodesQuery(Query):
+    name = "delete_isolated_nodes_query"
+    type = QueryType.WRITE
+    insert_return = False
+
+    async def query_init(self, db: InfrahubDatabase, **kwargs: dict[str, Any]) -> None:  # noqa: ARG002
+        query = """
+        MATCH p = (s)-[r]-(d)
```

Comment:
I htink it would be more efficient to have the label directly on the MATCH clause
```
MATCH p = (s:Node)-[r]-(d)
WHERE NOT exists((s)-[:IS_PART_OF]-(:Root))
```

Reply:
Yes definitely